scheduler-driver/src/test/java/gov/nasa/jpl/aerie/scheduler/TestApplyWhen.java

```diff
@@ -971,12 +971,13 @@ public void testCoexistenceWindowsCutoffMidActivity() {
     for(SchedulingActivityDirective a : plan.get().getActivitiesByTime()){
       logger.debug(a.startOffset().toString() + ", " + a.duration().toString());
     }
-
+    assertEquals(10, plan.get().getActivitiesById().size());
     assertTrue(TestUtility.activityStartingAtTime(plan.get(), Duration.of(2, Duration.SECONDS), actTypeB));
     assertTrue(TestUtility.activityStartingAtTime(plan.get(), Duration.of(10, Duration.SECONDS), actTypeB));
     assertTrue(TestUtility.activityStartingAtTime(plan.get(), Duration.of(16, Duration.SECONDS), actTypeB));
+    assertTrue(TestUtility.activityStartingAtTime(plan.get(), Duration.of(23, Duration.SECONDS), actTypeB));
     assertTrue(TestUtility.activityStartingAtTime(plan.get(), Duration.of(25, Duration.SECONDS), actTypeB));
-    assertEquals(5, problem.getSimulationFacade().countSimulationRestarts());
+    assertEquals(6, problem.getSimulationFacade().countSimulationRestarts());
```

Comment:
Why did using a closed interval for activities cause the number of sim restarts to increase? This question also applies to L904

Reply:
Because as you can see, there is also a new activity created in both of these tests. That is because activity ending at 19+3 ends now into the [23,28] interval and a coexisting activity can be created. 
Same explanation for the 8+4 anchor activity falling in the [12...] interval in the other test.